Tests/Model/AbstractElementTest.php

```diff
@@ -0,0 +1,270 @@
+                'nested/path',
+                'own namespace' => 'namespace',
+                'own/path',
+                [
+                    'namespace' => 'own/path',
+                ],
+            ],
+        ];
+    }
+
+    /**
+     * @return array
+     */
+    public function getStringRepresentation()
+    {
+        return [
+            'empty attrs and empty value' => [
+                'attrs' => [],
+                'value' => [],
+                'name' => 'sitemap',
+                'expected' => '<sitemap/>',
+            ],
+            'empty attrs and string value' => [
+                'attrs' => [],
+                'value' => '<test-tag/>',
+                'name' => 'sitemap',
+                'expected' => '<sitemap><test-tag/></sitemap>',
+            ],
+            'empty attrs and nested values' => [
+                'attrs' => [],
+                'value' => [$this->createElementStub('test-tag'), $this->createElementStub('test-tag-2')],
+                'name' => 'sitemap',
+                'expected' => '<sitemap><test-tag/><test-tag-2/></sitemap>',
+            ],
+            'many attrs and empty value' => [
+                'attrs' => ['created-at' => '11.12.2015', 'last-modified' => '12.12.2015'],
+                'value' => [],
+                'name' => 'sitemap',
+                'expected' => '<sitemap created-at="11.12.2015" last-modified="12.12.2015"/>',
+            ],
+            'many attrs and string value' => [
+                'attrs' => ['created-at' => '11.12.2015', 'last-modified' => '12.12.2015'],
+                'value' => '<test-tag/>',
+                'name' => 'sitemap',
+                'expected' => '<sitemap created-at="11.12.2015" last-modified="12.12.2015"><test-tag/></sitemap>',
+            ],
+            'many attrs and nested values' => [
+                'attrs' => ['created-at' => '11.12.2015', 'last-modified' => '12.12.2015'],
+                'value' => [$this->createElementStub('test-tag'), $this->createElementStub('test-tag-2')],
+                'name' => 'sitemap',
+                'expected' => '<sitemap created-at="11.12.2015" last-modified="12.12.2015"><test-tag/><test-tag-2/></sitemap>',
+            ],
+        ];
+    }
+
+    public function testMustImplementElementInterface()
+    {
+        $this->assertInstanceOf(ElementInterface::class, $this->element);
+    }
+
+    public function testAddNamespaceMustReturnSelf()
+    {
+        $this->assertEquals($this->element, $this->element->addNamespace('test name', '/test/path'));
+    }
+
+    /**
+     * @dataProvider getNamespaces
+     *
+     * @param array $namespaces
+     * @param array $expectedNamespaces
+     */
+    public function testAddAndGetNamespaceShouldWorkTogether(array $namespaces, array $expectedNamespaces)
+    {
+        foreach ($namespaces as $namePath) {
+            list($name, $path) = $namePath;
+            $this->element->addNamespace($name, $path);
```

Comment:
@andrey1s I can't use `[$name => $path]` because this way I won't be able to test duplicated namespaces (line 36). I've removed `list` and used your first suggestion.